ivy_tests/test_ivy/test_functional/test_core/test_creation.py

```diff
@@ -2,159 +2,358 @@
-# ---------------#
-
 # arange()
+@given(
+    start = st.integers(min_value = 0), 
+    stop = st.integers(min_value = 0) | st.none(),
+    step = st.integers().filter(lambda x : True if x != 0 else False),
+    dtype = st.sampled_from(ivy_np.valid_int_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="arange"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_arange(
+    start,
+    stop,
+    step,
+    dtype,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    helpers.test_array_function(
+        dtype,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        fw,
+        "arange",
+        start = start,
+        stop = stop,
+        step = step,
+    )
+
+
 # asarray()
+@given(
+    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_int_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="asarray"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_asarray(
+    dtype_and_x,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    dtype, x = dtype_and_x
+    helpers.test_array_function(
+        dtype,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        fw,
+        "asarray",
+        object_in = np.asarray(x),
+    )
+
+
 # empty()
+@given(
+    array_shape = st.integers(0,5) | st.lists(st.integers(min_value = 0, max_value=10), min_size = 1, max_size = 5),
+    dtype = st.sampled_from(ivy_np.valid_int_dtypes),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="empty"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_empty(
+    array_shape,
+    dtype,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):  
+    #couldnt find a better way to generate multi length arrays
+    if isinstance(array_shape, list):
+        array_shape = tuple(array_shape)
+
+    instance_method = False
+    container = False
+    as_variable = False
+    rtol: float = 1e-03,
+    atol: float = 1e-06,
+    test_values = False,
+    
+    helpers.test_array_function(
+        dtype,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        fw,
+        "empty",
+        rtol,
+        atol,
+        test_values,
+        shape=array_shape
```

Comment:
you should also be providing the keyword arguments `dtype` and `device` here. `dtype` is generated inside the hypothesis `@given` call, and `device` is generated by the pytest wrapping code, and so you can simply add `device` as an argument to `test_empty` and the argument will be populated with a device string.